tests/components/velbus/test_config_flow.py

```diff
@@ -24,15 +26,38 @@ def init_config_flow(hass):
     assert result['type'] == data_entry_flow.RESULT_TYPE_FORM
     assert result['step_id'] == 'user'
 
-    result = await flow.async_step_user({
-        CONF_NAME: 'Velbus Test Serial', CONF_PORT: PORT_SERIAL})
-    assert result['type'] == data_entry_flow.RESULT_TYPE_FORM
-    assert result['errors'] == {'port': 'connection_failed'}
-
-    result = await flow.async_step_user({
-        CONF_NAME: 'Velbus Test TCP', CONF_PORT: PORT_TCP})
-    assert result['type'] == data_entry_flow.RESULT_TYPE_FORM
-    assert result['errors'] == {'port': 'connection_failed'}
+    with patch('homeassistant.components.velbus.config_flow.'
+               'VelbusConfigFlow._test_connection', return_value=True):
```

Comment:
Don't patch the test connection method. That method is part of the code that should be tested. Patch the velbus controller class to return a mock controller instance.

Reply:
Because most is happening in the velbus connector init I completely failed doing it that way.
If you have an example I'm happy to convert but I will need help here.

Reply:
```py
@pytest.fixture(name='controller')
def mock_controller():
    controller = Mock()
    with patch('velbus.Controller', return_value=controller):
        yield controller
```
If you want an exception raised you set an exception as side effect when patching instead of return value.

Reply:
thanks